third_party/zstd.BUILD

```diff
@@ -0,0 +1,36 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "zstd",
+    srcs = glob([
+        "lib/common/*.c",
+        "lib/common/*.h",
+        "lib/compress/*.c",
+        "lib/compress/*.h",
+        "lib/decompress/*.c",
+        "lib/decompress/*.h",
+        "lib/dictBuilder/*.c",
+        "lib/dictBuilder/*.h",
+    ]) + select({
+        "@platforms//os:windows": [],
+        "//conditions:default": glob(["lib/decompress/*.S"]),
+    }),
+    hdrs = [
+        "lib/zdict.h",
+        "lib/zstd.h",
+        "lib/zstd_errors.h",
+    ],
+    includes = ["lib"],
+    linkopts = ["-pthread"],
```

Comment:
`-pthread` needs to be passed as a copt as well. It's used for setting up macros related to thread support.

From `man gcc`:
```
       -pthread
           Define additional macros required for using the POSIX threads library.  You should use
           this option consistently for both compilation and linking.  This option is supported
           on GNU/Linux targets, most other Unix derivatives, and also on x86 Cygwin and MinGW
           targets.
```

We should also probably not pass `-pthread` on Windows. While it works, I think it's just because MSVC ignores unknown options and prints a warning instead.